clients/shared/append.go

```diff
@@ -21,10 +21,12 @@ func Append(dwh destination.DataWarehouse, tableData *optimization.TableData, op
 		return fmt.Errorf("failed to get table config: %w", err)
 	}
 
+	cols, _ := columns.RemoveDeleteColumnMarker(tableConfig.Columns().GetColumns())
+
 	// We don't care about srcKeysMissing because we don't drop columns when we append.
 	_, targetKeysMissing := columns.Diff(
-		tableData.ReadOnlyInMemoryCols(),
-		tableConfig.Columns(),
+		tableData.ReadOnlyInMemoryCols().GetColumns(),
+		cols,
 		tableData.TopicConfig().SoftDelete,
```

Comment:
Can we hardcode `softDelete` here to be false? It should never be a soft delete for append

Reply:
Yep!